core/trino-spi/src/test/java/io/trino/spi/predicate/TestTupleDomain.java

```diff
@@ -112,6 +114,21 @@ public void testIntersection()
 
         assertEquals(tupleDomain1.intersect(tupleDomain2), expectedTupleDomain);
         assertEquals(tupleDomain2.intersect(tupleDomain1), expectedTupleDomain);
+
+        assertEquals(TupleDomain.intersect(ImmutableList.of()), all());
+        assertEquals(TupleDomain.intersect(ImmutableList.of(tupleDomain1)), tupleDomain1);
+        assertEquals(TupleDomain.intersect(ImmutableList.of(tupleDomain1, tupleDomain2)), expectedTupleDomain);
```

Comment:
yikes, we had no tests here for `TupleDomain.intersect(List)`... thanks for adding them!